cpp/src/gandiva/function_registry.cc

```diff
@@ -453,6 +72,17 @@ FunctionRegistry::SignatureMap FunctionRegistry::InitPCMap() {
     // printf("%s -> %s\n", entry->signature().ToString().c_str(),
     //      entry->pc_name().c_str());
   }
+#endif
+
+  FunctionRegistryArithmetic::GetArithmeticFnSignature(&map);
```

Comment:
since the Fns are adding the the map, can you pls name them more appropriately - maybe, AddFunctionSignatures() ?

Reply:
or, actually return a vector so that the knowledge of the specific map we are using isn't spread across all these files.

Reply:
changed it as per suggestion to use a vector.